docs/community/release-checklist.md

```diff
@@ -0,0 +1,340 @@
+# Release Checklist
+
+> If your experience deviates from this document, please document the changes to keep it up-to-date.
+
+This guide will guide you (pun intended) on your journey to release a new version of Krustlet.
+
+All releases will be of the form `vX.Y.Z` where `X` is the major version number, `Y` is the minor
+version number and `Z` is the patch release number. This project strictly follows [semantic
+versioning](https://semver.org/) so following this step is critical.
+
+It is important to note that this document assumes that the git remote in your repository that
+corresponds to <https://github.com/deislabs/krustlet> is named `upstream`. If yours is not (for
+example, if you've chosen to name it `origin` or something similar instead), be sure to adjust the
+listed snippets for your local environment accordingly.
+
+If you are not sure what your upstream remote is named, use a command like `git remote -v` to find
+out.
+
+If you don't have an upstream remote, you can add one easily using something like:
+
+```console
+git remote add upstream git@github.com:deislabs/krustlet
+```
+
+We are also going to be adding security and verification of the release process by providing
+signature files. We perform this using [GitHub and
+GPG](https://help.github.com/en/github/authenticating-to-github/about-commit-signature-verification).
+
+If you do not have GPG already setup you can follow these steps:
+
+1. [Install the GNU privacy Guard (GPG)](https://gnupg.org)
+1. [Generate a new GPG
+   key](https://help.github.com/en/github/authenticating-to-github/generating-a-new-gpg-key)
+1. [Add your key to your GitHub
+   account](https://help.github.com/en/github/authenticating-to-github/adding-a-new-gpg-key-to-your-github-account)
+1. [Set your signing key in
+   Git](https://help.github.com/en/github/authenticating-to-github/telling-git-about-your-signing-key)
+
+## Step 0: determine what type of release you are cutting
+
+Major releases are for new feature additions and behavioral changes that break backwards
+compatibility. Minor releases are for new feature additions that do not break backwards
+compatibility. Patch releases are for bug fixes that do not introduce any new features.
+
+There are two versions of the checklist: one for patch releases, and one for major/minor releases.
+Follow the checklist that best applies below.
+
+## Checklist for major and minor releases
+
+Follow this checklist if you are cutting a major release (vX.0.0) or a minor release (vX.Y.0):
+
+1. set up your environment
+1. increment the version number
+1. create the release branch
```

Comment:
Is this something we want to start doing now? I figured we'd start doing this after we hit 1.0

Reply:
yeah. Most of this is copied over from the Helm release checklist. Many of the items in this list don't apply. We can add these steps after we hit 1.0. I'll remove those steps for now.